contrib/download-frozen-image-v2.sh

```diff
@@ -72,30 +72,11 @@ fetch_blob() {
 	shift
 	local curlArgs=("$@")
 
-	local curlHeaders
-	curlHeaders="$(
-		curl -S "${curlArgs[@]}" \
-			-H "Authorization: Bearer $token" \
-			"$registryBase/v2/$image/blobs/$digest" \
-			-o "$targetFile" \
-			-D-
-	)"
-	curlHeaders="$(echo "$curlHeaders" | tr -d '\r')"
-	if grep -qE "^HTTP/[0-9].[0-9] 3" <<< "$curlHeaders"; then
-		rm -f "$targetFile"
-
-		local blobRedirect
-		blobRedirect="$(echo "$curlHeaders" | awk -F ': ' 'tolower($1) == "location" { print $2; exit }')"
-		if [ -z "$blobRedirect" ]; then
-			echo >&2 "error: failed fetching '$image' blob '$digest'"
-			echo "$curlHeaders" | head -1 >&2
-			return 1
-		fi
-
-		curl -fSL "${curlArgs[@]}" \
-			"$blobRedirect" \
-			-o "$targetFile"
-	fi
+	curl -L -S "${curlArgs[@]}" \
```

Comment:
We can use `-fsSL` now then I think? (like the other ones); doing so prevents capturing garbage if there's a failure (`-f` does a "fail fast" - preventing output to be captured on failure, and `-s` runs "silent" (but prints errors if they happen due to the other ones).

Reply:
(fine for a follow-up though)

Reply:
Should probably also drop the `-D-` to not dump headers to `stdout`.

Reply:
Yes, I was eyeing some of that as well, and got confused "are we capturing `stdout` here, or are we outputting to a file?

But it was a long day, and this fixed the immediate problem, so I left that for "let's look into that later; we now know the cause of the problem, and it's for our own CI only, so let's fix that first".


(contributions welcome though if you want to dig in! <3)